Update docs for workflows

ttung opened the pull request:
### Description
This encompasses the ingest, transform, and align workflows.

### Test plan
Visual inspection.

phoenixAja commented on docs/backend/workflows.md:
```diff
@@ -0,0 +1,19 @@
+
+### Transform GISAID
+
+This workflow applies the [transform-gisaid script](https://github.com/nextstrain/ncov-ingest/blob/master/bin/transform-gisaid) to transform the data into an input format suitable for the nextstrain pipeline.  The input for this workflow is a [`RawGisaidDump`](https://github.com/chanzuckerberg/aspen/search?q=%22class+RawGisaidDump%22) object and records a [`ProcessedGisaidDump`](https://github.co ...
```
can you expand a bit more on what suitable means here?

ttung replied:
I think it is just the metadata format.  I will clarify.

phoenixAja reviewed (approved):
awesome this is great!!! thanks for adding this, one last request is can you document the phylo workflow / how to kick it off (and what inputs/outputs it takes in/ creates)?

ttung commented:
> awesome this is great!!! thanks for adding this, one last request is can you document the phylo workflow / how to kick it off (and what inputs/outputs it takes in/ creates)?

I just added it, pls comment if needed. :)

Commit:
Update docs for workflows

### Description
This encompasses the ingest, transform, and align workflows.

### Test plan
Visual inspection.

sidneymbell reviewed (approved):
This is super helpful, thank you Tony!

phoenixAja commented:
> > awesome this is great!!! thanks for adding this, one last request is can you document the phylo workflow / how to kick it off (and what inputs/outputs it takes in/ creates)?
> 
> I just added it, pls comment if needed. :)

looks great!! 👍
